src/Akka.Persistence.Sql/Journal/SqlWriteJournal.cs

```diff
@@ -38,70 +38,96 @@ public static DateTime FromUnixEpochMillis(in long unixEpochMillis)
             => UnixEpoch.AddMilliseconds(unixEpochMillis);
     }
 
-    public class SqlWriteJournal : AsyncWriteJournal
+    public class SqlWriteJournal : AsyncWriteJournal, IWithUnboundedStash
     {
         [Obsolete(message: "Use SqlPersistence.DefaultConfiguration or SqlPersistence.Get(ActorSystem).DefaultConfig instead")]
         public static readonly Configuration.Config DefaultConfiguration = SqlPersistence.DefaultConfiguration;
 
-        private readonly ByteArrayJournalDao _journal;
+        private ByteArrayJournalDao _journal;
         private readonly JournalConfig _journalConfig;
         private readonly ILoggingAdapter _log;
 
-        private readonly ActorMaterializer _mat;
+        private ActorMaterializer _mat;
 
         private readonly Dictionary<string, Task> _writeInProgress = new();
+        private readonly CancellationTokenSource _pendingWriteCts;
+
+        public IStash Stash { get; set; }
 
         public SqlWriteJournal(Configuration.Config journalConfig)
         {
             _log = Context.GetLogger();
+            _pendingWriteCts = new CancellationTokenSource();
+
+            var config = journalConfig.WithFallback(SqlPersistence.DefaultJournalConfiguration);
+            _journalConfig = new JournalConfig(config);
+        }
+
+        protected override void PreStart()
+        {
+            base.PreStart();
+            Initialize().PipeTo(Self);
+            BecomeStacked(Initializing);
```

Comment:
Need to add a comment about why we need to use `BecomeStacked` here and not `Become`

Reply:
Done